google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/mutaterows/MutateRowsBatchingDescriptorV2.java

```diff
@@ -0,0 +1,127 @@
+
+/**
+ * A custom implementation of a {@link BatchingDescriptor} to split Rpc response into individual
+ * rows response and in a {@link MutateRowsException}.
+ *
+ * <p>This class is considered an internal implementation detail and not meant to be used by
+ * applications directly.
+ */
+// TODO(rahulkql): Consider renaming this class
+@InternalApi("For internal use only")
+public class MutateRowsBatchingDescriptorV2
+    implements BatchingDescriptor<RowMutationEntry, Void, BulkMutation, Void> {
+
+  @Override
+  public BatchingRequestBuilder<RowMutationEntry, BulkMutation> newRequestBuilder(
+      BulkMutation prototype) {
+    return new MyRequestBuilder(prototype);
+  }
+
+  @Override
+  public void splitResponse(Void som, List<SettableApiFuture<Void>> batch) {
+    for (SettableApiFuture<Void> batchResponse : batch) {
+      batchResponse.set(null);
+    }
+  }
+
+  @Override
+  public void splitException(Throwable throwable, List<SettableApiFuture<Void>> batch) {
+    if (!(throwable instanceof MutateRowsException)) {
+      for (SettableApiFuture<Void> future : batch) {
+        future.setException(throwable);
+      }
+      return;
+    }
+
+    List<FailedMutation> failedMutations = ((MutateRowsException) throwable).getFailedMutations();
+
+    Map<Integer, FailedMutation> errorsByIndex =
+        Maps.uniqueIndex(
+            failedMutations,
+            new Function<FailedMutation, Integer>() {
+              @Override
+              public Integer apply(@Nullable FailedMutation input) {
+                return input.getIndex();
+              }
+            });
+
+    int i = 0;
+    for (SettableApiFuture<Void> response : batch) {
+      // NOTE: The gax batching api doesn't allow for a single response to get different
+      // exception for different entries. However this does not affect this client because
+      // SettableApiFuture corresponds to  single element. So just use the last error per entry.
+      ApiException lastError = null;
+      FailedMutation failure = errorsByIndex.get(i++);
+
+      if (failure != null) {
+        lastError = failure.getError();
+      }
+
+      if (lastError == null) {
+        response.set(null);
+      } else {
+        response.setException(lastError);
+      }
+    }
+  }
+
+  /** {@inheritDoc} */
+  public long countBytes(RowMutationEntry request) {
+    return request.toProto().getSerializedSize();
+  }
+
+  /** A {@link BatchingRequestBuilder} that can aggregate BulkMutation */
+  static class MyRequestBuilder implements BatchingRequestBuilder<RowMutationEntry, BulkMutation> {
+    private BulkMutation bulkMutation;
+
+    MyRequestBuilder(BulkMutation prototype) {
+      this.bulkMutation = prototype.clone();
+    }
+
+    @Override
+    public void add(RowMutationEntry rowEntry) {
+      MutateRowsRequest.Entry entry = rowEntry.toProto();
+      ByteString rowK = entry.getRowKey();
+      bulkMutation.add(rowK, Mutation.fromProtoUnsafe(entry.getMutationsList()));
```

Comment:
Can BulkMutation take an instance of RowMutationEntry?

Reply:
Added a `BulkMutation#add(RowMutationEntry)`, I hope that should be fine.